pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1597,12 +1597,12 @@ public void checkTopicNsOwnership(final String topic) throws BrokerServiceExcept
     }
 
     public void cleanUnloadedTopicFromCache(NamespaceBundle serviceUnit) {
-        topics.forEach((name, topicFuture) -> {
-            TopicName topicName = TopicName.get(name);
+        for (String topic : topics.keys()) {
```

Comment:
does this `keys()` return a copy of the collection or is it a live cursor ?
should we make a copy here ?

Reply:
It returns a copy